geo-types/src/multi_point.rs

```diff
@@ -91,6 +98,78 @@ impl<T: CoordinateType> MultiPoint<T> {
     }
 }
 
+impl<T: CoordinateType> MultiPoint<T> {
+    /// Return the number of coordinates in the `MultiPoint`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use geo_types::MultiPoint;
+    ///
+    /// let mut coords = vec![(0., 0.), (5., 0.), (7., 9.)];
+    /// let multi_point: MultiPoint<f32> = coords.into_iter().collect();
+    /// assert_eq!(3, multi_point.num_coords());
+    /// ```
+    pub fn num_coords(&self) -> usize {
+        self.0.len()
+    }
+}
+
+#[cfg(test)]
+impl<T> RelativeEq for MultiPoint<T>
+where
+    T: CoordinateType + Float + AbsDiffEq,
+{
+    #[inline]
+    fn default_max_relative() -> Self::Epsilon {
+        T::epsilon()
+    }
+
+    #[inline]
+    fn relative_eq(
+        &self,
+        other: &MultiPoint<T>,
+        epsilon: Self::Epsilon,
+        max_relative: Self::Epsilon,
+    ) -> bool {
+        if self.num_coords() != other.num_coords() {
+            return false;
+        }
+
+        let mp_zipper = self.iter().zip(other.iter());
+        for (lhs, rhs) in mp_zipper {
+            if lhs.relative_ne(&rhs, epsilon, max_relative) {
+                return false;
+            }
+        }
+        true
+    }
+}
+
+#[cfg(test)]
+impl<T> AbsDiffEq for MultiPoint<T>
+where
+    T: CoordinateType + Float,
+{
+    type Epsilon = T;
+
+    #[inline]
+    fn default_epsilon() -> Self::Epsilon {
+        T::epsilon()
+    }
+
+    #[inline]
+    fn abs_diff_eq(&self, other: &MultiPoint<T>, epsilon: Self::Epsilon) -> bool {
```

Comment:
Can we do the `if self.num_coords() != other.num_coords() {` check?

Reply:
Thanks fixed. Also added test_abs_diff_eq() which is almost a copy of test_relative_eq() because that,  mistake of mine, should have been caught by the tests.